Tools/clinic/clinic.py

```diff
@@ -1433,19 +1438,21 @@ def render_option_group_parsing(self, f, template_dict):
                 continue
 
             group_ids = {p.group for p in subset}  # eliminate duplicates
-            d = {}
+            d: dict[str, str | int] = {}
             d['count'] = count
             d['name'] = f.name
             d['format_units'] = "".join(p.converter.format_unit for p in subset)
 
-            parse_arguments = []
+            parse_arguments: list[Any] = []
             for p in subset:
                 p.converter.parse_argument(parse_arguments)
             d['parse_arguments'] = ", ".join(parse_arguments)
 
             group_ids.discard(0)
-            lines = [self.group_to_variable_name(g) + " = 1;" for g in group_ids]
-            lines = "\n".join(lines)
+            lines = "\n".join([
+                self.group_to_variable_name(g) + " = 1;"
+                for g in group_ids
+            ])
```

Comment:
Could use a generator expression here rather than a list comprehension:

```suggestion
            lines = "\n".join(
                self.group_to_variable_name(g) + " = 1;"
                for g in group_ids
            )
```

Reply:
Yeah, but that's slower ;) The list approach generates faster code.

Reply:
Here I would prefer to go with the cleaner code rather than micro-optimise based on an implementation detail of the current algorithm used by `str.join` ;)

But I won't push this!